application/src/main/java/org/togetherjava/tjbot/logging/discord/DiscordLogForwarder.java

```diff
@@ -0,0 +1,154 @@
+import org.togetherjava.tjbot.logging.LogMarkers;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+/**
+ * Forwards log events to a Discord channel via a webhook. See {@link #forwardLogEvent(LogEvent)}.
+ * <p>
+ * Logs are forwarded in correct order, based on their timestamp. They are not forwarded
+ * immediately, but at a fixed schedule in batches of {@value MAX_BATCH_SIZE} logs.
+ * <p>
+ * Although unlikely to hit, the class maximally buffers {@value MAX_PENDING_LOGS} logs until
+ * discarding further logs. Under normal circumstances, the class can easily handle high loads of
+ * logs.
+ * <p>
+ * The class is thread-safe.
+ */
+final class DiscordLogForwarder {
+    private static final Logger logger = LoggerFactory.getLogger(DiscordLogForwarder.class);
+
+    private static final int MAX_PENDING_LOGS = 10_000;
+    private static final int MAX_BATCH_SIZE = WebhookMessage.MAX_EMBEDS;
+    private static final ScheduledExecutorService SERVICE =
+            Executors.newSingleThreadScheduledExecutor();
+    /**
+     * Has to be small enough for fitting all {@value MAX_BATCH_SIZE} embeds contained in a batch
+     * into the total character length of ~6000.
+     */
+    private static final int MAX_EMBED_DESCRIPTION = 1_000;
+    private static final Map<Level, Integer> LEVEL_TO_AMBIENT_COLOR =
+            Map.of(Level.TRACE, 0x00B362, Level.DEBUG, 0x00A5CE, Level.INFO, 0xAC59FF, Level.WARN,
+                    0xDFDF00, Level.ERROR, 0xBF2200, Level.FATAL, 0xFF8484);
+
+    private final WebhookClient webhookClient;
+    /**
+     * Internal buffer of logs that still have to be forwarded to Discord. Actions are synchronized
+     * using {@link #pendingLogsLock} to ensure thread safety.
+     */
+    private final Queue<LogMessage> pendingLogs = new PriorityQueue<>();
+    private final Object pendingLogsLock = new Object();
+
+    DiscordLogForwarder(URI webhook) {
+        webhookClient = WebhookClient.withUrl(webhook.toString());
+
+        SERVICE.scheduleWithFixedDelay(this::processPendingLogs, 5, 5, TimeUnit.SECONDS);
+    }
+
+
+    /**
+     * Forwards the given log message to Discord.
+     * <p>
+     * Logs are not immediately forwarded, but on a schedule. If the maximal buffer size of
+     * {@value MAX_PENDING_LOGS} is exceeded, logs are discarded.
+     * <p>
+     * This method is thread-safe.
+     *
+     * @param event the log to forward
+     */
+    void forwardLogEvent(LogEvent event) {
+        if (pendingLogs.size() >= MAX_PENDING_LOGS) {
+            logger.warn(LogMarkers.NO_DISCORD,
+                    """
+                            Exceeded the max amount of logs that can be buffered. \
+                            Logs are forwarded to Discord slower than they pile up. Discarding the latest log...""");
```

Comment:
I feel like it should be logged if the logs are starting to pile up already, before logs start to get discarded

Reply:
Thats fair and makes sense. I just consider all of this a really low priority. The logs in Discord arent supposed to be a point of truth.

Reply:
While I understand that the logs aren't supposed to be a point of truth.
I still don't want to just "discard logs randomly"

Even better would be, to log *on Discord* stating that logs are getting piled up. 

This way it actually gets noticed.

Reply:
It gets noticed by mods regardless, since we have warning-alerting on the VPS itself. But yes, this is a good suggestion.

Reply:
fixed in #607 by sending a warning log message when reaching a threshold of `8_000` (hard limit is `10_000`)